torch/_dynamo/variables/distributed.py

```diff
@@ -152,6 +152,42 @@ def var_getattr(self, tx, name: str) -> VariableTracker:
             return ConstantVariable.create(self.value.ndim)
         return super().var_getattr(tx, name)
 
+    def call_method(
+        self,
+        tx,
+        name,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        if name == "size":
```

Comment:
no need to do in this PR, but I think the logics for each methods are very ad hoc, we should directly call the method instead of letting dynamo mimic the method calls, as it's not quite stable and can easily break if the actual method in eager changes.


As a follow up PR, I think we should do sth like:
```
args_as_value = [x.as_python_constant() for x in args]
kwargs_as_value = {k: v.as_python_constant() for k, v in kwargs.items()}
# call the method and wrap with ConstantVariable
...
```

Reply:
At the very least, we should enable MyPy checking in the variables sub folder. It would catch a lot of bugs of undefined functions and methods.

Reply:
args_as_value = [x.as_python_constant() for x in args]
kwargs_as_value = {k: v.as_python_constant() for k, v in kwargs.items()}
# call the method and wrap with ConstantVariable
...

Agree in this case for .size() its probably all constants.  But we can't blindly assume all args/kwargs should be converted to python constants, sometimes they would be tensorvariables/symints too.

And yes, would be better to call self.value.size() and wrap in constant.

otoh, are we sure it's constant?  shouldn't we be checking for dynamic size here?

Reply:
Agree this will help make the code clean and generic, let me update it in next commit.

As far as I know, `.size()` is constant, and process groups are decided during initialization. I wonder if we have any cases that could make it dynamic?

Reply:
@yanboliang Should inlining be easy with your recent tracing rules work?